openedx/core/djangoapps/safe_sessions/middleware.py

```diff
@@ -501,6 +516,46 @@ def _delete_cookie(request, response):
         )
 
 
+def log_request_user_changes(request):
+    """
+    Instrument the request object so that we log changes to the `user` attribute. This is done by
+    changing the `__class__` attribute of the request object to point to a new class we created
+    on the fly which is exactly the same as the underlying request class but with an override for
+    the `__setattr__` function to catch the attribute chages.
+    """
+    original_user = getattr(request, 'user', None)
+
+    class SafeSessionRequestWrapper(request.__class__):
+        """
+        A wrapper class for the request object.
+        """
+
+        def __setattr__(self, name, value):
+            nonlocal original_user
+            if name == 'user' and hasattr(value, 'id'):
+                stack = inspect.stack()
+                location = "\n".join("%30s : %s:%d" % (t[3], t[1], t[2]) for t in stack[1:2])
+
+                if not hasattr(request, name):
+                    original_user = value
+                    log.info(
+                        f"SafeCookieData: Setting for the first time: '{value.id}'\n"
+                        f"{location}"
+                    )
+                elif value != getattr(request, name):
+                    current_user = getattr(request, name)
+                    log.info(
+                        f"SafeCookieData: Changing request user. "
+                        f"Originally '{original_user.id}', now '{current_user.id}' and will become '{value.id}'\n"
+                        f"{location}"
+                    )
+                else:
+                    # Value being set but not actually changing.
+                    pass
```

Comment:
`else: pass` is unusual.

Reply:
Yea, I just wanted to comment on what the case is that I'm not doing anything for explicitly for ease of readability.